### PR TITLE
Upgrade ipa  support

### PR DESCRIPTION
**Pull request by ccssu**
### Environment
Please Refer to the Readme in the Respective Repositories for Installation Instructions.

- ComfyUI:
  - github: https://github.com/comfyanonymous/ComfyUI
  - commit: 2d4164271634476627aae31fbec251ca748a0ae0 
  - Date:   Wed May 15 02:40:06 2024 -0400
 
  
- ComfyUI_IPAdapter_plus:
  - github: https://github.com/cubiq/ComfyUI_IPAdapter_plus
  - commit 20125bf9394b1bc98ef3228277a31a3a52c72fc2 
  - Date:   Wed May 8 16:10:20 2024 +0200

- ComfyUI_InstantID:
  - github: https://github.com/cubiq/ComfyUI_InstantID
  - commit d8c70a0cd8ce0d4d62e78653674320c9c3084ec1 
  - Date:   Wed May 8 16:55:55 2024 +0200

- OneDiff:
  - github: https://github.com/siliconflow/onediff 

    ```shell
    # install onediff
    git clone https://github.com/siliconflow/onediff.git
    cd onediff && git checkout dev_support_ipadapter_v2
    pip install -e .
    
    # install onediff_comfy_nodes
    ln -s $(pwd)/onediff_comfy_nodes path/to/ComfyUI/custom_nodes/
    # or
    # cp -r onediff_comfy_nodes path/to/ComfyUI/custom_nodes/
    ```

### CODE REVIEW

File: onediff_comfy_nodes/modules/oneflow/hijack_ipadapter_plus/set_model_patch_replace.py [hunk 1]
```diff
@@ -1,59 +1,153 @@
...
+    #     "cond": cond,
+    #     "cond_alt": cond_alt,
+    #     "uncond": uncond,
+    #     "weight_type": weight_type,
+    #     "mask": attn_mask,
+    #     "sigma_start": sigma_start,
+    #     "sigma_end": sigma_end,
+    #     "unfold_batch": unfold_batch,
+    #     "embeds_scaling": embeds_scaling,
+    # }
```
Comment (strint) [inline onediff_comfy_nodes/modules/oneflow/hijack_ipadapter_plus/set_model_patch_replace.py:47]: 这种可以直接删掉

File: onediff_comfy_nodes/modules/oneflow/hijack_ipadapter_plus/set_model_patch_replace.py [hunk 2]
```diff
@@ -1,59 +1,153 @@
...
+#         to["patches_replace"]["attn2"][key] = patch
+#     else:
+#         patch = to["patches_replace"]["attn2"][key]
+#         patch.set_new_condition(**torch2oflow(patch_kwargs))
+#         patch.set_cache(cache_key, len(patch.weights) - 1)
+#         patch.append_mask(masks_dict, patch_kwargs["mask"])
+
+#         if patch.get_bind_model() is not None:
+#             bind_model: CrossAttentionPatch_PT = patch.get_bind_model()
+#             bind_model.set_new_condition(**patch_kwargs)
```
Comment (strint) [inline onediff_comfy_nodes/modules/oneflow/hijack_ipadapter_plus/set_model_patch_replace.py:150]: 这种可以直接删掉

File: onediff_comfy_nodes/modules/oneflow/infer_compiler_registry/register_comfy/CrossAttntionStateDictPatch.py [hunk 3]
```diff
@@ -13,53 +12,32 @@ def process_state_dict_before_saving(state_dict: Dict, graph=None):
...
-                id_index_map = {
-                    id(t): i for i, t in enumerate(graph._state_tensor_tuple)
-                }
-                name_idx = {
-                    f"{graph_key}-{k}-{attn_k}": id_index_map[id(attn_t)]
-                    for k, attn_m in attn2_patches.items()
-                    for attn_k, attn_t in attn_m.state_dict().items()
-                    if id(attn_t) in id_index_map
-                }
+                # id_index_map = {
```
Comment (strint) [inline onediff_comfy_nodes/modules/oneflow/infer_compiler_registry/register_comfy/CrossAttntionStateDictPatch.py:15]: 这个现在不需要了？

File: onediff_comfy_nodes/modules/oneflow/patch_management/patch_executor.py [hunk 4]
```diff
@@ -106,3 +106,34 @@ def copy_to(self, old_model: ModelPatcher, new_model: ModelPatcher):
...
+        """UNetExtraInputOptions
+       
+        """
+        super().__init__()
+        self.patch_name = type(self).__name__
+
+    def check_patch(self, module):
+        return hasattr(module, self.patch_name)
+
+    def set_patch(self, module, value: Dict):
```
Comment (strint) [inline onediff_comfy_nodes/modules/oneflow/patch_management/patch_executor.py:121]: 这个就是转 module 为 input 的方法么

File: onediff_comfy_nodes/modules/oneflow/hijack_ipadapter_plus/README.md [hunk 5]
```diff
@@ -4,24 +4,25 @@ Please Refer to the Readme in the Respective Repositories for Installation Instr
...
+  - commit d8c70a0cd8ce0d4d62e78653674320c9c3084ec1 
+  - Date:   Wed May 8 16:55:55 2024 +0200
+  
 - ComfyUI-AnimateDiff-Evolved$ git log
   - commit f9e0343f4c4606ee6365a9af4a7e16118f1c45e1 
   - Date:   Sat Apr 6 17:32:15 2024 -0500
 
 - OneDiff:
   - github: https://github.com/siliconflow/onediff 
+  - branch:
```
Comment (jackalcooper) [inline onediff_comfy_nodes/modules/oneflow/hijack_ipadapter_plus/README.md:25]: 漏写了吗

File: onediff_comfy_nodes/modules/oneflow/patch_management/patch_executor.py (hunk 4 above)
Comment (ccssu) [reply onediff_comfy_nodes/modules/oneflow/patch_management/patch_executor.py:121]: 是的

File: onediff_comfy_nodes/modules/oneflow/infer_compiler_registry/register_comfy/CrossAttntionStateDictPatch.py [hunk 6]
```diff
@@ -13,53 +12,20 @@ def process_state_dict_before_saving(state_dict: Dict, graph=None):
...
-                                attn_key, graph._state_tensor_tuple[idx]
-                            )
-                        
-    except Exception as e:
-        print(f"Warning: Failed to apply patch after loading state dict: {e}")
+def attn2_patch_sum(input_kwargs)->int:
+    attn2 = input_kwargs.get("transformer_options", {}).get("patches_replace", {}).get(
+        "attn2", {}
+    )
+    return len(attn2)
```
Comment (strint) [inline onediff_comfy_nodes/modules/oneflow/infer_compiler_registry/register_comfy/CrossAttntionStateDictPatch.py:31]: 需要 format 一下

**Review by strint — APPROVED** [review]

File: onediff_comfy_nodes/modules/oneflow/infer_compiler_registry/register_comfy/CrossAttntionStateDictPatch.py (hunk 3 above)
Comment (ccssu) [reply onediff_comfy_nodes/modules/oneflow/infer_compiler_registry/register_comfy/CrossAttntionStateDictPatch.py]: 不需要了

**Pull request merged**